optaplanner-core/src/main/java/org/optaplanner/core/impl/solver/termination/BasicPlumbingTermination.java

```diff
@@ -98,14 +105,43 @@ public synchronized boolean waitForRestartSolverDecision() {
         }
     }
 
+    /**
+     * Concurrency note: unblocks {@link #waitForRestartSolverDecision()}.
+     *
+     * @param problemChange never null
+     * @return as specified by {@link Collection#add}
+     */
+    public synchronized boolean addProblemChange(ProblemChange<Solution_> problemChange) {
```

Comment:
I'm not too excited about the fact that we have both `addPC()` and `addPFC()`.
This termination is not public API - we can afford to change the API and use only one entry point for this.

Reply:
True, it might already accept the adapter type.